File: selfstudy/python/students_filterhigh.py
'''
task 115:
Students - FilterHigh
You need to return only the values greater than 50 from student test scores
'''

def above_fifty_students(scores):
    return [x for x in scores if x > 50]

# Tests
if __name__ == '__main__':
    tests = [
        ([10, 20, 30], above_fifty_students([10, 20, 30])),
        ([50, 60, 70], above_fifty_students([50, 60, 70])),
        ([1, 2, 3], above_fifty_students([1, 2, 3])),
        ([100, 200], above_fifty_students([100, 200])),
        ([], above_fifty_students([]))
    ]
    import sys
    if 'pytest' in sys.argv:
        # pytest structure
        pass
    else:
        for idx, (inp, exp) in enumerate(tests):
            res = above_fifty_students(inp)
            print(f"Test {idx+1}:", "PASS" if res == exp else "FAIL")

# done

File: selfstudy/golang/students_filterhigh.go
/*
task 215:
Students - FilterHigh
You need to return only the values greater than 50 from student test scores
*/

package main

import (
	"fmt"
	"reflect"
)

func Above_fiftyStudents(scores []int) []int {
	var res []int
	for _, v := range scores {
		if v > 50 { res = append(res, v) }
	}
	return res
}

func main() {
	tests := []struct {
		input []int
		expected []int
	}{
		{[]int{10, 20, 30}, Above_fiftyStudents([]int{10, 20, 30})},
		{[]int{50, 60, 70}, Above_fiftyStudents([]int{50, 60, 70})},
		{[]int{1, 2, 3}, Above_fiftyStudents([]int{1, 2, 3})},
		{[]int{100, 200}, Above_fiftyStudents([]int{100, 200})},
		{[]int{}, Above_fiftyStudents([]int{})},
	}
	
	for idx, t := range tests {
		res := Above_fiftyStudents(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// done
